### PR TITLE
chore(git): clean up .gitignore files

### PR DESCRIPTION
**Pull request by sigveio**
## Description 

This pull request aims to re-organise the projects `.gitignore` to help keep it clean, and to add various common editor generated folders/files to the list. It also removes some redundant lines from the site/docusaurus `.gitignore`, which are covered by the root.

The pattern `*.bundle.js` is also deleted.
_(From what I could gather this was added ~5 years ago for a compiler that is now deprecated and no longer used)_

## Motivation / Context

I generated a `.vscode/launch.json` file for the vscode debugger while working on a PR, and thought it would be practical for developers to avoid having to stash or delete these types of files between commits. So I wanted to suggest adding common ignores for the most popular IDEs. Which seemed like a good opportunity for some housekeeping =)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
+yarn-error.log*
+lerna-debug.log*
+.pnpm-debug.log*
 
-# Cache
+# Editor
+.idea
+.vscode
+.history
+*.suo
```
Comment (ludofischer) [inline .gitignore:35]: I have never seen these extensions:
* `.ntvs*`
* `.njsproj`
* `.sln`
* `.sw?` 

What are they?

File: .gitignore [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
-dist
-lerna-debug.log
+# Package manager
 node_modules
-npm-debug.log
-yarn-error.log
+
+# Build
+dist
 _webpackOutput
```
Comment (ludofischer) [inline .gitignore:6]: Do you think `_webpackOutput` is still needed? What would generate this file? The only occurrence on Google for `_webpackOutput` is... this file.

**Review by ludofischer — COMMENTED** [review]
Did you gather the list from somewhere or is it based on your own usage only?

File: .gitignore (hunk 1 above)
Comment (sigveio) [reply .gitignore:35]: They are mostly related to Visual Studio. The two first are related to their Node.js tools. And the `.sln` is (like `*.suo`) related to [organising projects](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/solution-dot-sln-file) there.

`.sw?` takes care of vim temporary files

File: .gitignore (hunk 2 above)
Comment (sigveio) [reply .gitignore:6]: Hmm, good catch. I somehow just assumed it was something meaningful when I saw it was webpack related - but I can't find any external references to it anywhere either. 

From the git blame it seems to have been added by Ben 5 years ago

```bash
❯ grep -rnw cssnano -e '_webpackOutput'
(...)
cssnano/packages/cssnano/src/__tests__/_webpack.config.js:8:    path: path.resolve('./_webpackOutput/'),
```

Test related?

**Comment by sigveio** [comment]
> Did you gather the list from somewhere or is it based on your own usage only?

I boiled it together from a combination of my own standard `.gitignore` / dotfiles and the existing ones in this project. I can't recall where I originally gathered these code editor related extensions from - some are from own usage, and some probably inspired from other `.gitignores` I have come across.

File: .gitignore [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
+.idea
+.vscode
+.history
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
 
-.eslintcache
```
Comment (alexander-akait) [inline .gitignore:21]: Better to keep it, because eslint generate cache

Comment (sigveio) [reply .gitignore:21]: It's just been moved to another location in the file, under the `# Cache` section :-)

File: .gitignore (hunk 2 above)
Comment (ludofischer) [reply .gitignore:6]: You're right, it's a file that the tests create.

**Review by ludofischer — APPROVED** [review]

**Pull request merged**